analysis/quantifier-comparisons/pre-processing-benchmarking-metrics.Rmd

```diff
@@ -0,0 +1,612 @@
+  tidyr::drop_na()
+```
+
+For simplicity, let's just look at correlation of each tool with cellranger
+```{r}
+cell_qc_common_cor %>% 
+  dplyr::group_by(sample) %>%
+  dplyr::summarize(
+    cr_al_cor = cor(cellranger, alevin, method = "spearman"),
+    cr_al_fry_cor =cor(cellranger, alevin_fry, method = "spearman"),
+    cr_al_fry_sketch_cor = cor(cellranger, alevin_fry_sketch, method = "spearman"),
+    cr_ka_cor = cor(cellranger, kallisto, method = "spearman")
+  )
+```
+Overall, they all have really high correlations with cellranger, although it does look like Alevin-fry has the consistently highest correlation with kallisto consistently the lowest correlation. 
+
+```{r}
+ggplot(cell_qc_common_cor, aes(x = cellranger, y = alevin_fry)) +
+  geom_point(size = 0.5, alpha = 0.1) + 
+  facet_wrap(~ sample) + 
+  scale_x_log10() + 
+  scale_y_log10() + 
+  labs(x = "Cell Ranger UMI/cell", y = "Alevin Fry UMI/cell") + 
+  theme_classic()
+```
+
+What about looking at the correlation of genes/cell. I think this should be relatively the same as UMI/cell. 
+```{r}
+gene_qc_common_cor <- cell_qc_common %>%
+  # spread the mean expression stats to one column per caller
+  tidyr::pivot_wider(id_cols = c(cell_id, sample),
+                     names_from = tool,
+                     values_from = detected) %>%
+  # drop rows with NA values to ease correlation calculations
+  tidyr::drop_na()
+```
+
+```{r}
+gene_qc_common_cor %>% 
+  dplyr::group_by(sample) %>%
+  dplyr::summarize(
+    cr_al_cor = cor(cellranger, alevin, method = "spearman"),
+    cr_al_fry_cor =cor(cellranger, alevin_fry, method = "spearman"),
+    cr_al_fry_sketch_cor = cor(cellranger, alevin_fry_sketch, method = "spearman"),
+    cr_ka_cor = cor(cellranger, kallisto, method = "spearman")
+  )
+```
+There is some slight variation, but still quite hgih. Again, it looks like Alevin fry has the best, with Kallisto having the worst. 
+
+```{r}
+ggplot(gene_qc_common_cor, aes(x = cellranger, y = alevin_fry)) +
+  geom_point(size = 0.5, alpha = 0.1) + 
+  facet_wrap(~ sample) + 
+  scale_x_log10() + 
+  scale_y_log10() + 
+  labs(x = "Cell Ranger Gene/cell", y = "Alevin Fry Gene/cell") + 
+  theme_classic()
+```
+```{r}
+ggplot(gene_qc_common_cor, aes(x = cellranger, y = kallisto)) +
+  geom_point(size = 0.5, alpha = 0.1) + 
+  facet_wrap(~ sample) + 
+  scale_x_log10() + 
+  scale_y_log10() + 
+  labs(x = "Cell Ranger Gene/cell", y = "Kallisto Gene/cell") + 
+  theme_classic()
+```
+When actually looking at each gene, you see that kallisto starts to show some variation, where as alevin fry has a much cleaner diagonal and is more similar to cellranger for the cells that are detected by both. Another thing to note, which is shown in the boxplots previously, the ranges for these comparisons are not always the same. So yes, there is strong correlation, but cellranger still has a lower range of gene/cell than both Alevin-fry and kallisto.   
+
+### Per Gene QC
```

Comment:
It probably doesn't matter too much, but maybe we want to separate this into a different notebook? Obviously there is shared info here, but for clarity it might be worth doing. Maybe.